reactivesocket-cpp/src/tcp/TcpDuplexConnection.h

```diff
@@ -0,0 +1,93 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <folly/ExceptionWrapper.h>
+#include <folly/io/async/AsyncSocket.h>
+#include <folly/io/async/AsyncTransport.h>
+#include <reactivesocket-cpp/src/streams/SmartPointers.h>
+#include "reactivesocket-cpp/src/DuplexConnection.h"
+#include "reactivesocket-cpp/src/Payload.h"
+#include "reactivesocket-cpp/src/ReactiveStreamsCompat.h"
+#include "reactivesocket-cpp/src/mixins/IntrusiveDeleter.h"
+#include <folly/SocketAddress.h>
+#include <folly/io/IOBufQueue.h>
+
+namespace folly {
+    class exception_wrapper;
+}
+
+namespace reactivesocket {
+    class TcpSubscriptionBase: public virtual ::reactivesocket::IntrusiveDeleter,
+                               public ::reactivesocket::Subscription {
+    public:
+        ~TcpSubscriptionBase() = default;
+
+        // Subscription methods
+        void request(size_t n) override {
+            LOG(INFO) << "TcpSubscriptionBase.request " << n;
```

Comment:
in the TODO comments, I would add
+ implement proper request(n) logic.

it looks like you are ignoring it for now.